helpdesk_mgmt/controllers/myaccount.py

```diff
@@ -65,7 +65,13 @@ def portal_my_tickets(
         searchbar_filters = {
             "all": {"label": _("All"), "domain": []},
         }
-        for stage in request.env["helpdesk.ticket.stage"].search([]):
+        for stage in request.env["helpdesk.ticket.stage"].search(
+            [
+                "|",
+                ("team_ids", "=", HelpdeskTicket.team_id.id),
+                ("team_ids", "=", False),
```

Comment:
You can do it in one shot:

```suggestion
                ("team_ids", "in", [False] + HelpdeskTicket.team_id.ids),
```

But anyway, that's incorrect. `HelpdeskTicket` is an empty recordset, so this is never going to have a team. You should have the team elsewhere.

Reply:
This suggestion is still not applied and I don't see a central method to get this. Something like `def _get_applicable_stages(self, team):` or similar.

Reply:
I defined _get_applicable_stages in portal and in helpdesk.ticket in order to get central method. Tell me if you see it okay now, thanks :)